indico/web/forms/fields/principals.py

```diff
@@ -21,15 +21,32 @@
 from wtforms import HiddenField
 
 from indico.core.db.sqlalchemy.principals import PrincipalType
+from indico.modules.events.roles.util import serialize_role
 from indico.modules.groups import GroupProxy
 from indico.modules.groups.util import serialize_group
 from indico.modules.networks.models.networks import IPNetworkGroup
 from indico.modules.networks.util import serialize_ip_network_group
 from indico.modules.users.util import serialize_user
 from indico.util.user import principal_from_fossil
+from indico.web.forms.fields import JSONField
 from indico.web.forms.widgets import JinjaWidget
 
 
+def serialize_principal(principal):
+    if principal.principal_type == PrincipalType.email:
+        return principal.fossilize()
```

Comment:
Let's avoid calling legacy stuff here! Eventually we want to get rid of fossilize. You could just add a `serialize_email_principal` function and maybe even see if you can get rid of the fossilization stuff in `EmailPrincipal` (it seems to be used in just one other place).

Reply:
Actually it looks like the "one other place" is the one you removed just below ;)